### ISSUE TITLE
Organization and structure of the Oceania data server

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmt#1645]
**Simplify maintenance of the GMT data server**

GMT users access remote files and Earth relief grids from https://oceania.generic-mapping-tools.org/cache and https://oceania.generic-mapping-tools.org, respectively. There is no HTML there so one cannot browse to these directories; we use libcurl calls to obtain files.  The top directory also has _gmt_hash_server.txt_, the list with files and their SHA256 hash and sizes.  GMT refreshes this file on a user's computer once a day (at the most) and use that information to know if an already downloaded file needs to be deleted and downloaded anew before use.

At present, there are ~175 cache files and 13 global relief grids on the server.  The cache files are a motley crew of mostly small test files for the documentation, examples, tests, animations, and the tutorial.  They follow no naming convention and presumably these files can come and go as we need them in the examples of scripts.  Perhaps with one exception, all the cache files are small enough to fit in a GitHub repository.  At the present, they are actually kept in an old subversion repo (gmt-cache).

The Earth relief grids are likely to always be derivatives of one or more original grids.  These can always be recomputed and do not need to be in a repo; however, the recipes should be in the repo.  In the future, it is likely other data sources for other planets will join Earth.

If we create a repo called _gmtserver-admin_ or similar we could have gmtserver do a git pull nightly (or more often) via crontab to ensure the server is current; any of us can also do this manually as needed.  Perhaps we could have a symbolic link _cache_ that points to the dataserver-admin/cache folder to avoid having to duplicate the files?

I have written and tested a script that lists all the files in cache plus the grids, then for each file determines if it is younger than the current hash table or not.  If it is, we compute new sha256 and sizes values and write an updated hash record, otherwise we just grep the existing record.  At the end the hash table is overwritten with the new file.  This runs very fast since, apart from setup, we don't expect files to change very frequently.  Thus, we could have that script run in crontab quite frequently (every 10 minutes?) so that nobody actually has to do anything other than make pull requests from their local machine

The derived data grids are different: Here we need a script that detects that an original grid has changed (i.e., younger than the derived products) and that should trigger rebuilds of all the derived products.  We need to use GMT for this, so gmtserver should have a working GMT installation on it (updated via cron as well - I do that on some other machines here).  It seems we may need a table of info for each original dataset that specifies which resolutions to build and any other technical details so a general script can run on any source.

While this can be close to what we need right now, should the NASA proposal that Leo and I submitted this summer be funded then we will want to break down the larger relief files into tiles (like SRTM but larger) so that users don't have to download 2.6 Gb such as the 15s grid just to make a map of France. We won't do tiling until later.

Hoping for comments from @GenericMappingTools/core.

**Comment by PaulWessel** [comment]
I have created a new private repository called gmtserver-admin which has a cache directory with all the files except the 138 Mb topo.8.2.img.  Suggestions for handling this file is welcome. Too big for git.  We could give a full URL but then it gets downloaded twice (two tests use it).

**Comment by seisman** [comment]
Is it possible to resample topo.8.2.img to lower resolution?

**Comment by PaulWessel** [comment]
Yes, but not sure if a generic tool can do this.  The file is simply a headerless binary short integer file, so if we could use dd or some other tool to skip every 2 bytes then we could dumb it down to 4 m.  Or I write a 2 line C program...

**Comment by PaulWessel** [comment]
OK, wrote that program and have topo.8.4.img and will see how it goes.  May have to change the script since the code is now looking for a 4m version so may have to pass size.

**Comment by PaulWessel** [comment]
Closing as completed for now.  More issues can be listed under the gmtserver-admin repo.
